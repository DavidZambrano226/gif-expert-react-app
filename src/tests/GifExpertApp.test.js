import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpextApp';

describe('GifExpert App tests', () => {
    
    let wrapper;

    beforeEach( () => {
        wrapper = shallow(<GifExpertApp />);
    })

    test('should make match with the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should render a list of the Gif Grid categories', () => {
        const categories = ['Cats', 'Dogs', 'Smile'];

        const defaultWrapper = shallow(<GifExpertApp defaultCategories={categories} />)


        expect( defaultWrapper ).toMatchSnapshot();
        expect( defaultWrapper.find('GifGrid').length ).toBe( categories.length );

    })
    
    
})
