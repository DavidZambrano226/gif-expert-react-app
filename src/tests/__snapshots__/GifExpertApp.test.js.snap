// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpert App tests should make match with the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Animals"
      key="Animals"
    />
  </ol>
</Fragment>
`;

exports[`GifExpert App tests should render a list of the Gif Grid categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Cats"
      key="Cats"
    />
    <GifGrid
      category="Dogs"
      key="Dogs"
    />
    <GifGrid
      category="Smile"
      key="Smile"
    />
  </ol>
</Fragment>
`;
